lnwire/message_test.go

```diff
@@ -0,0 +1,844 @@
+package lnwire_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"image/color"
+	"io"
+	"math"
+	"math/big"
+	"math/rand"
+	"net"
+	"sync"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tor"
+	"github.com/stretchr/testify/require"
+)
+
+const deliveryAddressMaxSize = 34
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+var (
+	testSig = &btcec.Signature{
+		R: new(big.Int),
+		S: new(big.Int),
+	}
+	testNodeSig, _ = lnwire.NewSigFromSignature(testSig)
+
+	testNumExtraBytes = 1000
+	testNumSigs       = 100
+	testNumChanIDs    = 1000
+	buffer            = make([]byte, 0, lnwire.MaxSliceLength)
+
+	bufPool = sync.Pool{
+		New: func() interface{} {
+			return bytes.NewBuffer(buffer)
+		},
+	}
+)
+
+// BenchmarkWriteMessage benchmarks the performance of lnwire.WriteMessage. It
+// generates a test message for each of the lnwire.Message, calls the
+// WriteMessage method and benchmark it.
+func BenchmarkWriteMessage(b *testing.B) {
+	// Create testing messages. We will use a constant seed to make sure
```

Comment:
If you add `b.ReportAllocs` here, then it'll also show information related to the number of allocations/bytes per op.

Reply:
Got it. I used `-benchmem` instead, so `go test -run=^$ -bench=Write -benchmem` will give,
```
BenchmarkWriteMessage/Init-4    545257    2330 ns/op    176 B/op    9 allocs/op
```